operator/src/main/java/oracle/kubernetes/utils/OperatorUtils.java

```diff
@@ -19,4 +25,84 @@ public static String joinListGrammatically(final List<String> list) {
             .concat(list.get(list.size() - 1))
         : list.get(0);
   }
+
+  /**
+   * Create a Map using the elements from the given map, with their keys sorted
+   * using the 'numero lexi sorting name'.
+   *
+   * @param map Map containing elements to be sorted by the keys
+   * @param <T> Type of map entries
+   * @return A sorted Map containing the elements from the give map
+   */
+  public static <T> Map<String, T> createSortedMap(Map<String, T> map) {
+    if (map == null) {
+      return Collections.emptyMap();
+    }
+    return map.entrySet()
+        .stream()
+        .sorted(Comparator
+            .comparing((Entry<String, T> entry) -> getSortingString(entry.getKey())))
+        .collect(Collectors.toMap(
+            Map.Entry::getKey,
+            Map.Entry::getValue,
+            (oldValue, newValue) -> oldValue, LinkedHashMap::new));
+  }
+
+  /**
+   * Compare the 'numero lexi sorting name' as defined in {@link #getSortingString(String)} of the
+   * given 2 Strings.
+   *
+   * @param str1 First string for comparison
+   * @param str2 Second string for comparison
+   * @return a negative integer, zero, or a positive integer as the sorting name of str1 * is less
+   *     than, equal to, or greater than the sorting name of str2.
+   */
+  public static int compareSortingStrings(String str1, String str2) {
+    if (str1 == null || str2 == null) {
+      if (str2 != null) {
+        return -1;
+      } else if (str1 != null) {
+        return 1;
+      }
+      return 0;
+    }
+    return getSortingString(str1).compareTo(getSortingString(str2));
+  }
+
+  /**
```

Comment:
i guess i would question if we really need to implement a sort algorithm???

Reply:
did you write this, or copy it from somewhere? if you copied it, we need attribution, and we need to check licenses and copyrights.

Reply:
This is new and it's definitely needed.

The domain status was listing information in a random order, and the rolls were in a random order.

Reply:
It comes from pull OWLS-80522 Domain status ordering, OWLS-80144 Ordered rolling upgrade.  https://github.com/oracle/weblogic-kubernetes-operator/pull/1503

Reply:
https://jira.oraclecorp.com/jira/browse/OWLS-80944